Firedoors will crush mobs when closing Part 2

Shadowtail117 opened the pull request:
Re-opening to add the removal of the obstruction checks just in case #28266 fails. Original PR was https://github.com/Baystation12/Baystation12/pull/28268.

----------
Functionally the same as airlocks crushing things. Damage is slightly increased compared to airlocks since firedoors slam shut rather than use actuators to slowly close.

Do note that this also removes the check for if it would crush something when closing. Ultimately, the firedoors exist to stop the spread of a hostile atmosphere, and if it means crushing someone (which can be treated easily) or compromising a large portion of the ship's atmosphere (which cannot be treated easily), I personally would rather it do the former.

After discussion, firedoors will now give you a warning before closing so you aren't caught unawares.

:cl:
tweak: Firedoors (emergency shutters) will now crush mobs that are under them when they close. Better stay clear.
tweak: Firedoors will no longer refuse to close if something is on top of their tile.
tweak: Firedoors will now give two seconds of warning before closing to give you time to get out of the way.
/:cl:

Shadowtail117 committed:
Merge pull request #1 from Baystation12/dev

Update

Shadowtail117 committed:
Update

HeyBanditoz reviewed (changes requested):
At least let the user know they're going to get hurt.

Shadowtail117 commented:
You mean by giving them a notice saying "You just got crushed!"? I intend to make this similar to airlocks crushing you, with the added damage being the only exception, and airlocks offer no such niceties.

HeyBanditoz commented:
Doesn't closing a shutter produce a dialog box

Shadowtail117 commented:
Only if you close it manually. But in that situation, you'd either be closing it on yourself, which makes no sense, or on someone else, in which case they wouldn't know. Firedoors automatically closing produce no dialog.

Shadowtail117 commented:
@HeyBanditoz See latest commit.

HeyBanditoz commented on code/game/machinery/doors/firedoor.dm:
```diff
@@ -348,12 +346,13 @@
-
 	latetoggle()
+	visible_message("<span class='danger'>The firedoor beeps ominously, get out of the way!</span>", "You hear some beeping coming from the ceiling.", 3)
+	sleep(10)
```
Use a [timer](https://github.com/Baystation12/Baystation12/wiki/Timers) instead.

Shadowtail117 replied:
This is intended to block close() since it's adding 1 second of delay before the door actually begins closing. Why do you want a timer?

HeyBanditoz replied:
Nvm oopa I'm an idiot I read #cosing just njiw

Shadowtail117 committed:
Firedoors will now crush mobs when closing

Firedoors now give 1 second of warning

Only damage mobs

Span macros

Change 1 second delay to 2 seconds

Change 20 to 2 SECONDS
